src/riak_kv_pb_timeseries.erl

```diff
@@ -293,6 +296,156 @@ process(SQL = #riak_sql_v1{'FROM' = Table}, State) ->
             {reply, missing_helper_module(Table, BucketProps), State};
         DDL ->
             submit_query(DDL, Mod, SQL, State)
+    end;
+%% No support yet for replacing coverage components, so replace_cover
+%% must be undefined
+process(#tscoveragereq{query = Q,
+                       replace_cover=undefined,
+                       table = Table}, State) ->
+    Mod = riak_ql_ddl:make_module_name(Table),
+    Bucket = table_to_bucket(Table),
+    SQL = compile(Mod, catch decode_query(Q)),
+    Client = {riak_client, [node(), undefined]},
+
+    case SQL of
+        {error, _Error} ->
+            {reply, make_rpberrresp(?E_BAD_QUERY,
+                                    "Failed to compile query"),
+             State};
+        _ ->
+            %% SQL is a list of queries (1 per quantum)
+            convert_cover_list(sql_to_cover(Client, SQL, Bucket, []), State)
+    end.
+
+%% Copied and modified from riak_kv_pb_coverage:convert_list. Would
+%% be nice to collapse them back together, probably with a closure,
+%% but time and effort.
+convert_cover_list({error, Error}, State) ->
+    {error, Error, State};
+convert_cover_list(Results, State) ->
+    %% Pull hostnames & ports
+    %% Wrap each element of this list into a rpbcoverageentry
+    Resp = #tscoverageresp{
+              entries=
+                  lists:map(fun({Cover, Range, SQLtext}) ->
```

Comment:
I would add a private helper function called `coverageentry_from_result/1` instead of the anonymous function.